samples/test/datasets.test.js

```diff
@@ -105,4 +111,34 @@ describe('Datasets', () => {
     const [exists] = await bigquery.dataset(datasetId).exists();
     assert.strictEqual(exists, false);
   });
+
+  // Only delete a resource if it is older than 24 hours. That will prevent
+  // collisions with parallel CI test runs.
+  function isResourceStale(creationTime) {
+    const oneDayMs = 86400000;
+    const now = new Date();
```

Comment:
Another approach you could take here is to use `Date.now()`.  If `creationTime` is  a "milliseconds since bleh" kind of value:
```js
return Date.now() - creationTime >= oneDayMs
```